tools/test-proxy/documentation/asset-sync/README.md

```diff
@@ -25,49 +25,61 @@ An `assets.json` takes the form:
-| AssetsRepoBranch | The branch within the assets repo that your updated recordings will be pushed to. |
-| SHA | The reference SHA the recordings that should be restored from the assets repository. |
+| AssetsRepoPrefixPath | The assets repository may want to place the content under a specific path in the assets repo. The default is the language that the assets belong to. EG: `python`, `net`, `java` etc. |
+| TagPrefix | `<Language>/<ServiceDirectory>` or `<Language>/<ServiceDirectory>/<Library>` or deeper if things are nested in such a manner. |
+| Tag | Initially empty until after the first push at which point the tag will be the `<TagPrefix><Guid>` |
 
 Comments within the assets.json are allowed and _maintained_ by the tooling. Feel free to leave notes to yourself. They will not be eliminated.
 
 As one can see in the example image above, the test-proxy does the heavy lifting for push and pull of files to and from the assets repository.
 
-### Restore, push, reset when proxy is waiting for requests
+## Restore, push, reset when proxy is waiting for requests
 
 Interactions with the external assets repository are accessible when the proxy is actively serving requests. These are available through routes:
 
 | Route | Description |
 |---|---|
-| `/Playback/Restore` | Retrieve files from external git repo as targeted in the SHA from assets.json |
-| `/Playback/Reset` | Discard pending changes and reset to the original SHA from targeted assets.json. |
-| `/Record/Push` | Push changes if they are pending for files described by targeted assets.json. |
+| `/Playback/Restore` | Retrieve files from external git repo as targeted in the Tag from assets.json |
+| `/Playback/Reset` | Discard pending changes and reset to the original Tag from targeted assets.json. |
+| `/Record/Push` | Push changes if they are pending for files described by targeted assets.json. Updates the Tag in assets.json the new tag generated from the push operation. |
 
-### Restore, push, reset as a CLI app
+## test-proxy CLI commands
 
-The test-proxy also offers interactions with the external assets repository as a CLI app. What this means is that one could invoke
+The test-proxy also offers interactions with the external assets repository as a CLI. Invoking `test-proxy --help` will show the available list of commands. `test-proxy <command> --help` will show the help and options for an individual command. The options for a given command are all `--<option>`, for example, `--assets-json-path`, but each option has an abbreviation shown in the help, those are a single dash. For example the abbreviation for `--assets-json-path` is `-a`.
+
+### The following CLI commands are available for manipulation of assets
+
+#### Restore
+
+Restore will pull the assets at the Tag, from the AssetsRepo referenced by the targeted assets.json
```

Comment:
```suggestion
A restore operation is merely a test-proxy-encapsulated `clone or pull` operation. A given `asets.json` provides the target `Tag` and `AssetsRepo`.
```

I don't like this really either, but I don't like what it is now.

Reply:
I'll update with the next commit.